blog_post_collaboration/spark_streaming_dataset.md

```diff
@@ -0,0 +1,367 @@
+# A new Kedro dataset for Spark Structured Streaming
+
+This article describes a new [Kedro dataset for Spark Structured Streaming](https://github.com/kedro-org/kedro-plugins/tree/main/kedro-datasets/kedro_datasets/spark). [Spark Structured Streaming](https://spark.apache.org/docs/latest/structured-streaming-programming-guide.html) is built on the Spark SQL engine. You can express your streaming computation the same way you would express a batch computation on static data, and the Spark SQL engine will run it incrementally and continuously and update the final result as streaming data continues to arrive. 
+
+## Introduction
+
+This needs to be written:
+* What this post solves for the reader, and who would be interested in learning about this.
+* Limitations of the dataset (if any). Proof of concept only??
+* Link to source code and further information about Kedro datasets.
+* What the following sections show: set up a project, configure the data catalog
+
+## Set up a project to use the `SparkStreaming` Kedro dataset
+
+These are the steps to follow in order to use the dataset:
+
+The methodology for integrating Kedro with Spark streaming for real-time processing of data streams involves the following steps:
+
+1. Create a Kedro project using the `pyspark` starter
+2. Extend the template project
+3. Register the necessary PySpark and streaming related Hooks
+4. Configure the custom dataset in the catalog.yml file, defining the streaming sources and sinks.
+5. Use Kedro’s MemoryDataSet to store intermediate dataframes generated during the Spark streaming process.
+
+By defining the sources and sinks in the catalog.yml file and using the MemoryDataSet, Kedro creates a structured data pipeline that can read and write data streams with Spark streaming. This allows for efficient processing of data streams in real-time.
+
+
+### Create a Kedro project
+
+Create a Kedro project using the Kedro `pyspark` starter: 
+
+```bash
+kedro new --starter=pyspark`
+```
+
+WHAT VERSION OF KEDRO AND KEDRO-DATASETS??
+
+### Extend the project
+
+1. Create a new folder called `extras` in `src/$your_kedro_project_name/extras`. Within this folder, create a subfolder called `datasets`.
```

Comment:
Once we installed the latest version of kedro-datasets, this step can be skipped